templates/CRM/Contribute/Form/ContributionView.tpl

```diff
@@ -68,6 +68,11 @@
   </div>
 </div>
 <table class="crm-info-panel">
+  {if $is_test}
+    <div class="help">
+      <strong>{ts}This transaction was submitted using a TEST payment processor{/ts}</strong>
```

Comment:
It is not always true that every test transaction is submitted using Test payment processor, it can be pay later as well.